[NO JIRA] Update PR template (#2883)
Skyscanner/backpack: .github/pull_request_template.md

- [ ] Component `README.md`
- [ ] Tests
- [ ] Storybook examples created/updated

> - [ ] Migration guidelines added to the description of the PR. If the guide is lengthy, consider creating a new Markdown page inside the component's docs folder and link it here

Reviewer: ```suggestion
- [ ] For breaking changes, migration guidelines added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here
```

Author: I am just thinking that it's not exactly breaking changes if we follow a deprecation cycle? I guess as a concept it is, but it won't be released as a breaking change

Reviewer: I guess we can reword it to be, only adding this to provide the PR raiser context on when they should be providing migration guides otherwise it leaves it open to confusion/interpretation which might mean we have to reject a PR until we go back and ask for it

```suggestion
- [ ] For breaking changes or deprecating components/properties, migration guides added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here
```
